File: language-examples/c#/AbstractClasses.cs
using System;
using System.Collections.Generic;

namespace AbstractClasses
{
	public abstract class BaseClass
	{
		public abstract void Method();
	}

	public class SubClass1 : BaseClass
	{
		public override void Method()
		{
			Console.WriteLine("SubClass1 Method Called");
		}
	}

	public class SubClass2 : BaseClass
	{
		public override void Method()
		{
			Console.WriteLine("SubClass2 Method Called");
		}
	}

	public class SubClass3 : BaseClass
	{
		public override void Method()
		{
			Console.WriteLine("SubClass3 Method Called");
		}
	}

	public class SubClass4 : BaseClass
	{
		public override void Method()
		{
			Console.WriteLine("SubClass4 Method Called");
		}
	}

	class Program
	{
		static void Main()
		{
			List<BaseClass> list = new List<BaseClass>();

			list.Add(new SubClass1());
			list.Add(new SubClass2());
			list.Add(new SubClass3());
			list.Add(new SubClass4());

			foreach(var subClass in list)
			{
				subClass.Method();
			}
		}
	}
}

File: language-examples/java/AbstractClasses.java
import java.util.*;

abstract class AbstractClass
{
	public abstract void method();
}

class SubClass1 extends AbstractClass
{
	@Override
	public void method()
	{
		System.out.println("SubClass1 Method Called");
	}
}

class SubClass2 extends AbstractClass
{
	@Override
	public void method()
	{
		System.out.println("SubClass2 Method Called");
	}
}

class SubClass3 extends AbstractClass
{
	@Override
	public void method()
	{
		System.out.println("SubClass3 Method Called");
	}
}

class SubClass4 extends AbstractClass
{
	@Override
	public void method()
	{
		System.out.println("SubClass4 Method Called");
	}
}

public class AbstractClasses
{
	public static void main(String[] args)
	{
		List<AbstractClass> list = new ArrayList<AbstractClass>();

		list.add(new SubClass1());
		list.add(new SubClass2());
		list.add(new SubClass3());
		list.add(new SubClass4());

		for(AbstractClass subClass : list)
		{
			subClass.method();
		}
	}
}